simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -291,6 +345,63 @@ KOKKOS_FORCEINLINE_FUNCTION basic_simd<T, simd_abi::scalar> condition(
       static_cast<bool>(a) ? static_cast<T>(b) : static_cast<T>(c));
 }
 
+template <class T, class BinaryOperation>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
+reduce(Experimental::basic_simd<T, Experimental::simd_abi::scalar> const& x,
+       Experimental::basic_simd_mask<T, Experimental::simd_abi::scalar> const&
+           mask,
+       T identity, BinaryOperation) noexcept {
+  if (!mask) return identity;
+  return x[0];
+}
+
+template <class T, class BinaryOperation>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
+reduce(Experimental::basic_simd<T, Experimental::simd_abi::scalar> const& x,
+       BinaryOperation binary_op) noexcept {
+  return reduce(
+      x,
+      Experimental::basic_simd<T, Experimental::simd_abi::scalar>::mask_type(
+          true),
+      Impl::Identity<T, BinaryOperation>(), binary_op);
+}
+
+template <class T>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T reduce_min(
+    Experimental::basic_simd<T, Experimental::simd_abi::scalar> const& x,
+    Experimental::basic_simd_mask<T, Experimental::simd_abi::scalar> const&
+        mask) noexcept {
+  if (!mask) return Kokkos::reduction_identity<T>::min();
+  return x[0];
+}
+
+template <class T>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
+reduce_min(Experimental::basic_simd<T, Experimental::simd_abi::scalar> const&
+               x) noexcept {
+  return reduce_min(
+      x, Experimental::basic_simd<T, Experimental::simd_abi::scalar>::mask_type(
+             true));
+}
```

Comment:
Why not simply return x[0]? You call the above function with true, which return x[0]

Reply:
Yes, switched to simply return `x[0]`.